src/mots/config.py

```diff
@@ -0,0 +1,214 @@
+    """Thrown when a particular module is not valid."""
+
+    pass
+
+
+class FileConfig:
+    """Loader and writer for filesystem based configuration."""
+
+    def __init__(self, path: Path = DEFAULT_CONFIG_FILEPATH):
+        """Initialize the configuration with provided config file path."""
+        if not path.exists() and path.is_file():
+            raise ValueError(f"{path} does not exist or is not a file.")
+        self.path = path
+        self.repo_path = path.parent
+        self.config = None
+
+    def init(self):
+        """Initialize a repo with a config file, if it does not contain it."""
+        if not self.path.is_file():
+            # File does not exist, create it.
+            now = datetime.now().isoformat()
+            self.config = {
+                "repo": str(Path(self.path).resolve().parts[-2]),
+                "created_at": now,
+                "updated_at": None,
+                "people": [],
+                "modules": [],
+            }
+            self.write()
+            logger.info(f"mots configuration initialized in {self.path}.")
+        else:
+            logger.warning(f"mots configuration file detected in {self.path}.")
+
+    def load(self):
+        """Load configuration from file."""
+        with self.path.open("r") as f:
+            self.config = yaml.load(f)
+
+    def write(self):
+        """Write configuration to file, and update the timestamp."""
+        self.config["updated_at"] = datetime.now().isoformat()
+        with self.path.open("w") as f:
+            yaml.dump(self.config, f)
+
+
+def clean(file_config: FileConfig, write: bool = True):
+    """Clean and re-sort configuration file.
+
+    Load configuration from disk, sort modules and submodules by `machine_name`. If
+    there is no valid `machine_name`, generate one. Write changes to disk.
+
+    :param file_config: an instance of :class:`FileConfig`
+    :param write: if set to `True`, writes changes to disk.
+    """
+    people_keys = ("owners", "peers")
+    file_config.load()
+    directory = Directory(file_config)
+    directory.load()
+    for i, module in enumerate(file_config.config["modules"]):
+        if "machine_name" not in module:
+            module["machine_name"] = generate_machine_readable_name(module["name"])
+
+        for key in people_keys:
+            if key not in module or not module[key]:
+                continue
+            for i, person in enumerate(module[key]):
+                if person["bmo_id"] not in directory.people.by_bmo_id:
+                    file_config.config["people"].append(person)
+                    module[key][i] = person
+                    directory.people.refresh_by_bmo_id()
+                else:
+                    module[key][i] = file_config.config["people"][
+                        directory.people.by_bmo_id[person["bmo_id"]]
+                    ]
+
+        # Do the same for submodules.
+        # TODO: this should be refactored.
```

Comment:
Let's move this "`TODO`" to a bug.
If you wouldn't mind, can you pop a "find all" for "`TODO`" and move 'em all?

Reply:
[bug filed](https://bugzilla.mozilla.org/show_bug.cgi?id=1753753). Will replace the rest of TODOs with bugs, where needed.

Reply:
Note that there's still a couple `TODO`s in the tests, if you want to snatch those up too.